Sources/NIOCore/AsyncSequences/NIOAsyncWriter.swift

```diff
@@ -37,6 +37,18 @@ public protocol NIOAsyncWriterSinkDelegate: Sendable {
     /// - Important: You **MUST NOT** call ``NIOAsyncWriter/Sink/setWritability(to:)`` from within this method.
     func didYield(contentsOf sequence: Deque<Element>)
 
+    /// This method is called once a single element was yielded to the ``NIOAsyncWriter``.
+    ///
+    /// If the ``NIOAsyncWriter`` was writable when the sequence was yielded, the sequence will be forwarded
+    /// right away to the delegate. If the ``NIOAsyncWriter`` was _NOT_ writable then the sequence will be buffered
+    /// until the ``NIOAsyncWriter`` becomes writable again. All buffered writes, while the ``NIOAsyncWriter`` is not writable,
+    /// will be coalesced into a single sequence.
+    ///
+    /// - Note: This a fast path that you can optionally implement. By default this will just call ``NIOAsyncWriterSinkDelegate/didYield(contentsOf:)``.
+    ///
+    /// - Important: You **MUST NOT** call ``NIOAsyncWriter/Sink/setWritability(to:)`` from within this method.
+    func didYield(_ element: Element)
```

Comment:
I'm wondering if we can instead use a single `Deque` we reuse specifically for single yields. If users don't hold onto it we would only allocate once or CoW if they still hold it.

Reply:
I was thinking the same but that doesn't really work out. The problem in high throughput is that we are expecting users to hop onto the EL in their `didYield` implementation. At the same time the producing thread is calling `yield` again on the writer. Most of the times we are still holding onto the `Deque` in the EL thread at this point. This is the safer fast path that we can guarantee to hit.